cmd/k8s-operator/proxygroup.go

```diff
@@ -399,6 +398,34 @@ func (r *ProxyGroupReconciler) ensureConfigSecretsCreated(ctx context.Context, p
 			mak.Set(&cfgSecret.StringData, tsoperator.TailscaledConfigFileName(cap), string(cfgJSON))
 		}
 
+		// The config sha256 sum is used to trigger pod restarts when config
+		// changes. We only use the first pod's config as input because we
+		// don't want to restart all pods unnecessarily every time the
+		// ProxyGroup is scaled up or down. But you can't change the config in
+		// a way that one pod needs a restart without others needing a restart,
+		// so it should be sufficient to just hash one pod's config.
+		//
+		// In future, we're aiming to eliminate restarts altogether and have
+		// pods dynamically reload their config when it changes on disk.
```

Comment:
Suggestion (less verbose + so that it still makes sense in future when we don't mount all configs to all pods)

```suggestion

		// The config sha256 sum is a value for a hash annotation used to trigger pod restarts when tailscaled 
                //  config
		// changes. Any config changes apply to all replicas, so it is sufficient to only hash the config for the first 
                // replica.
		//
		// In future, we're aiming to eliminate restarts altogether and have
		// pods dynamically reload their config when it changes on disk.
```